src/hypothesis/settings.py

```diff
@@ -259,6 +266,52 @@ def __exit__(self, *args, **kwargs):
         default_context_manager = self.defaults_stack().pop()
         return default_context_manager.__exit__(*args, **kwargs)
 
+    @staticmethod
+    def register_profile(name, settings):
+        """registers a collection of values to be used as a settings profile.
+        These settings can be loaded in by name. Enable different defaults for
+        different settings.
+
+        - settings is a Settings object
+
+        """
+        Settings._profiles[name] = copy.copy(settings)
+
+    @staticmethod
+    def get_profile(name):
+        """Return the profile with the given name.
+
+        - name is a string representing the name of the profile
+         to load
+        A InvalidArgument exception will be thrown if the
+         profile does not exist
+
+        """
+        try:
+            return copy.copy(Settings._profiles[name])
+        except KeyError:
+            raise InvalidArgument(
+                "Profile '{}' has not been registered".format(
```

Comment:
Yeah, sorry. :-( I very much regret supporting Python 2.6 but each little change like this doesn't seem *quite* worth dropping it for.